src/windows.rs

```diff
@@ -15,107 +36,51 @@ pub struct Client {
 #[derive(Debug)]
 pub struct Acquired;
 
-type BOOL = i32;
-type DWORD = u32;
-type HANDLE = *mut u8;
-type LONG = i32;
-
-const ERROR_ALREADY_EXISTS: DWORD = 183;
-const FALSE: BOOL = 0;
-const INFINITE: DWORD = 0xffffffff;
-const SEMAPHORE_MODIFY_STATE: DWORD = 0x2;
-const SYNCHRONIZE: DWORD = 0x00100000;
-const TRUE: BOOL = 1;
-const WAIT_OBJECT_0: DWORD = 0;
-
-extern "system" {
-    fn CloseHandle(handle: HANDLE) -> BOOL;
-    fn SetEvent(hEvent: HANDLE) -> BOOL;
-    fn WaitForMultipleObjects(
-        ncount: DWORD,
-        lpHandles: *const HANDLE,
-        bWaitAll: BOOL,
-        dwMilliseconds: DWORD,
-    ) -> DWORD;
-    fn CreateEventA(
-        lpEventAttributes: *mut u8,
-        bManualReset: BOOL,
-        bInitialState: BOOL,
-        lpName: *const i8,
-    ) -> HANDLE;
-    fn ReleaseSemaphore(
-        hSemaphore: HANDLE,
-        lReleaseCount: LONG,
-        lpPreviousCount: *mut LONG,
-    ) -> BOOL;
-    fn CreateSemaphoreA(
-        lpEventAttributes: *mut u8,
-        lInitialCount: LONG,
-        lMaximumCount: LONG,
-        lpName: *const i8,
-    ) -> HANDLE;
-    fn OpenSemaphoreA(dwDesiredAccess: DWORD, bInheritHandle: BOOL, lpName: *const i8) -> HANDLE;
-    fn WaitForSingleObject(hHandle: HANDLE, dwMilliseconds: DWORD) -> DWORD;
-    #[link_name = "SystemFunction036"]
-    fn RtlGenRandom(RandomBuffer: *mut u8, RandomBufferLength: u32) -> u8;
-}
-
-// Note that we ideally would use the `getrandom` crate, but unfortunately
-// that causes build issues when this crate is used in rust-lang/rust (see
-// rust-lang/rust#65014 for more information). As a result we just inline
-// the pretty simple Windows-specific implementation of generating
-// randomness.
-fn getrandom(dest: &mut [u8]) -> io::Result<()> {
```

Comment:
The quoted issue is closed.

Is that still not ok?
If so, I can add this as a feature.

Reply:
The issue is closed because it was about diagnostics, afaict, but the issue is just a matter of life for rustc work, aiui. I think let's not add a feature and just use the crate to replace this as you've done, and if rustc wants to use this they can submit a PR to fix it up or something.

Reply:
Would update the README.md then.

Reply:
Done